src/handle.jl

```diff
@@ -33,6 +33,7 @@ immutable Handle{T <: AbstractSundialsObject}
     ptr_ref::Ref{Ptr{T}} # pointer to a pointer
 
     @compat function (::Type{Handle}){T}(ptr::Ptr{T})
+        T <: Void && throw(ArgumentError("Handle assigned to a null pointer"))
```

Comment:
Pointer comparison actually doesn't check the pointer type, just an address.
I guess, `T <: Void` would never be true in the context of `common.jl` as no `CreateXXX` return `Ptr{Void}`.
That's why having a `@test_throws` would be nice.